libraries/AP_HAL_ChibiOS/hwdef/common/common_eth.ld

```diff
@@ -0,0 +1,35 @@
+/*
+    ChibiOS - Copyright (C) 2006..2015 Giovanni Di Sirio
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+/*
+    this file is included by the board specific ldscript.ld which is
+    generated from hwdef.dat
+*/
+
+SECTIONS
+{
+    /* Special section for Ethernet DMA non cache-able areas.*/
+    .eth (NOLOAD) : ALIGN(4)
```

Comment:
why is this needed with the new malloc approach?

Reply:
Right now the dynamic malloc is a build option depending on STM32_ETH_BUFFERS_EXTERN. Without it this linker file (and changes to the memory map) are needed. If STM32_ETH_BUFFERS_EXTERN is going to be a requirement for Ardupilot in ChibiOS then the linker file is not needed. I'm in favor of removing it